src/EventHub/EventHub/Cmdlets/NetworkRuleSet/RemoveAzureEventHubIPRule.cs

```diff
@@ -0,0 +1,93 @@
+    public class RemoveAzureEventHubIPRule : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = IPRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [Parameter(Mandatory = true, ParameterSetName = IPRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = IPRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [Parameter(Mandatory = true, ParameterSetName = IPRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }        
+
+        [Parameter(Mandatory = true, ParameterSetName = IPRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Subnet Resource ID")]
+        [ValidateNotNullOrEmpty]
+        public string IpMask { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = IPRuleInputObjectParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "IPRule Configuration Object")]
+        [ValidateNotNullOrEmpty]
+        public PSNWRuleSetIpRulesAttributes IpRuleObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            PSNetworkRuleSetAttributes networkRuleSet = Client.GetNetworkRuleSet(ResourceGroupName, Name);
+
+            PSNWRuleSetIpRulesAttributes Toremove = new PSNWRuleSetIpRulesAttributes();
+
+            if (ParameterSetName.Equals(IPRuleInputObjectParameterSet))
+            {
+                Toremove = IpRuleObject;
+            }
+
+            if (ParameterSetName.Equals(IPRulePropertiesParameterSet))
+            {
+                foreach (PSNWRuleSetIpRulesAttributes iprule in networkRuleSet.IpRules)
+                {
+                    if (iprule.IpMask.Equals(IpMask))
+                    {
+                        Toremove = iprule;
+                    }
+                }
+            }
+            
+
+            if (Toremove != null)
+            {
+                if (ShouldProcess(target: Name, action: string.Format("Removing IPrule for NetworkRuleSet of {0} in Resourcegroup {1}", Name, ResourceGroupName)))
+                {
+                    try
+                    {
+                        //Add the IpRules
+                        networkRuleSet.IpRules.Remove(Toremove);
+                        Client.CreateOrUpdateNetworkRuleSet(ResourceGroupName, Name, networkRuleSet);
```

Comment:
@v-Ajnava for `Remove-*` cmdlets, the `-PassThru` parameter should be added to allow the user to receive output if they choose that they want it. You can read [this document](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#returning-no-output) for more information on returning no output in cmdlets.

Ideally, if this cmdlet is removing an entire resource, then you would return `true` if `-PassThru` is provided, and if this is removing a property on a resource, you would return the base object if `-PassThru` is provided.

Reply:
Resolved